logstash-core/lib/logstash/environment.rb

```diff
@@ -72,6 +72,7 @@ module Environment
            Setting::Boolean.new("help", false),
            Setting::Boolean.new("enable-local-plugin-development", false),
             Setting::String.new("log.format", "plain", true, ["json", "plain"]),
+           Setting::Boolean.new("log.format.json_strict", false),
```

Comment:
what do you think about having `json.strict` instead of `json_script`?

This opens the door to having sub properties of `json` or allowing `strict` for other log formats - I'm thinking ahead to ECS logging at some point

Reply:
The idea was to keep close the concept of strict (fixing, avoiding the duplication of keys) to the fact that was specific to json. Now I think that renaming it to `log.format.json.fix_duplicate_message_fields` is more appropriate, and stick with that definition.

Reply:
Fixed by 923756094db5c8c89bbfe2115567cf98606b94dd